packages/metadata/src/Decorated/storage/fromMetadata/createFunction.spec.ts

```diff
@@ -104,6 +104,15 @@ describe('createFunction', (): void => {
         (): Uint8Array => storageFn(['5DXUeE5N5LtkW97F2PzqYPyqNkxqSWESdGSPTX6AvkUAhwKP'])
       ).toThrow(/requires two arguments/);
     });
+
+    it('accepts an optional parameter for key construction', (): void => {
+      expect(storageFn.iterKey).not.toBeUndefined();
+
+      const result = storageFn.iterKey!('5DXUeE5N5LtkW97F2PzqYPyqNkxqSWESdGSPTX6AvkUAhwKP');
```

Comment:
No `!` please. (Trying to rid the codebase of them, sadly we have a handful - even in tests, tests are the actual reason we have not been able to switch on the rule, i.e. https://github.com/polkadot-js/api/blob/master/.eslintrc.js#L16)

Reply:
I'm not sure how I can do so, because the TypeScript compiler is not smart enough to know that I've already checked for it to be non-null or non-undefined in a statement above.